internal/validators/oci_validator.go

```diff
@@ -299,6 +316,34 @@ func setupTransportOpts(opts []remote.Option, caCert string) ([]remote.Option, e
 	return opts, nil
 }
 
+// verifySignature verifies the authenticity of the given image reference URL using the provided public keys.
+func verifySignature(ctx context.Context, ref name.Reference, pubKeys [][]byte, opt ...remote.Option) error {
+	ctxTimeout, cancel := context.WithTimeout(ctx, verificationTimeout)
+	defer cancel()
+
+	defaultCosignOciOpts := []soci.Options{
+		soci.WithRemoteOptions(opt...),
+	}
+
+	for _, key := range pubKeys {
+		verifier, err := soci.NewCosignVerifier(ctxTimeout, append(defaultCosignOciOpts, soci.WithPublicKey(key))...)
+		if err != nil {
+			return err
+		}
+
+		verified, err := verifier.Verify(ctxTimeout, ref)
+		if err != nil {
+			continue
```

Comment:
I'd recommend logging the error for debugging

Reply:
Added details and errs in this case.